constantine/serialization/io_limbs.nim

```diff
@@ -63,8 +63,17 @@ func unmarshalLE[T](
 
     # if full, dump
     if acc_len >= wordBitWidth:
-      if dst_idx == dst.len:
-        return false
+      if dst_idx == dst.len: # return, however check if all remaining
+        # elements are 0 (more significant bits beyond `src_idx`)
+        # For example, given a
+        # `src = "DDCCBBAA0000"`
+        # and a 4 byte LittleEndian `BigInt`, `src` is fine, because the most
+        # significant two bytes are all zero.
+        var allZero = true
+        for jidx in src_idx ..< dst.len:
+          if src[jidx] != 0: allZero = false
```

Comment:
I'll have to review more in-depth that case but we can't leak the position of non-zero bytes as this breaks constant-time property.
And even leaking if something is zero is tricky.